packages/flutter/lib/src/material/mergeable_material.dart

```diff
@@ -0,0 +1,507 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+/// A [Widget] that abstracts over a Material slice and a Material gap.
+abstract class MergeableMaterialItem extends StatelessWidget {
+  MergeableMaterialItem({Key key}) : super(key: key);
+
+  /// Return whether the current item is a [MaterialGap].
+  bool get isGap;
+}
+
+/// A [Widget] that can be used as a child to [MergeableMaterial]. It is a slice
+/// of [Material] that animated merges within the list.
+class MaterialSlice extends MergeableMaterialItem {
+  /// Creates a slice of [Material] that's mergeable within a
+  /// [MergeableMaterial].
+  MaterialSlice({
+    Key key,
+    this.child
+  }) : super(key: key);
+
+  /// The widget below this widget in the tree.
+  final Widget child;
+
+  @override
+  bool get isGap => false;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Material(
+      type: MaterialType.transparency,
+      child: child
+    );
+  }
+}
+
+/// A [Widget] that represents a gap within [MergeableMaterial].
+class MaterialGap extends MergeableMaterialItem {
+  /// Creates a Material gap with a given size.
+  MaterialGap({
+    Key key,
+    this.size: 16.0
+  }) : super(key: key);
+
+  /// The size of the gap, irrespective of the axis used.
+  final double size;
+
+  @override
+  bool get isGap => true;
+
+  // Dummy object; real Widget generated later.
+  @override
+  Widget build(BuildContext context) => null;
+}
+
+/// A [Widget] useful for animating the merging and separating of [Material]
```

Comment:
no need to hyperlink `[Widget]`, just say `widget`. Hyperlinking the second word of the definition detracts from the definition.